### PR TITLE
Issue #4718: Use localStorage to store tableDrag.showWeight value

### LINKED ISSUE #4718: Fixed: Add sameSite attribute to Backdrop tableDrag cookie.
When visiting a page with tabledrag, Firefox nags into the console:

`Cookie “Backdrop.tableDrag.showWeight” will be soon rejected because it has the “sameSite” attribute set to “none” or an invalid value, without the “secure” attribute.
`
More information about that nagging: https://developer.mozilla.org/de/docs/Web/HTTP/Headers/Set-Cookie/SameSite

Backdrop uses [jquery.cookie.js](https://github.com/backdrop/backdrop/blob/1.x/core/misc/jquery.cookie.js)

Actually using a cookie for tabledrag is not necessary it just toggles display stuff via js. I'd recommend to use [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) instead.

This is a related [Drupal 8 issue](https://www.drupal.org/project/drupal/issues/1569648), where they switched to localStorage as well.

### PR DESCRIPTION
**Pull request by klonos**
Fixes https://github.com/backdrop/backdrop-issues/issues/4718

Alternative to https://github.com/backdrop/backdrop/pull/3869
Mimics: https://git.drupalcode.org/project/drupal/-/commit/148871a8c914c2db9c28ceac3ebf9763a0e83e7e + follow-up https://git.drupalcode.org/project/drupal/-/commit/96bca83bcc2eb969d1330f84092652bc2615faf8

### CODE REVIEW

File: core/misc/tabledrag.js [hunk 1]
```diff
@@ -98,15 +113,10 @@ Backdrop.tableDrag = function (table, tableSettings) {
...
-    .click(function () {
-      if ($.cookie('Backdrop.tableDrag.showWeight') == 1) {
-        self.hideColumns();
-      }
-      else {
-        self.showColumns();
-      }
-      return false;
-    })
+    .click($.proxy(function (e) {
```
Comment (indigoxela) [inline core/misc/tabledrag.js]: I see one possible problem with this: [jQuery.proxy() is deprecated](https://api.jquery.com/jQuery.proxy/)

File: core/misc/tabledrag.js [hunk 2]
```diff
@@ -123,6 +133,15 @@ Backdrop.tableDrag = function (table, tableSettings) {
   $(document).on('touchend', function (event) { return self.dropRow(event.originalEvent.touches[0], self); });
   $(document).on('mousemove pointermove', function (event) { return self.dragRow(event, self); });
   $(document).on('mouseup pointerup', function (event) { return self.dropRow(event, self); });
+  // React to localStorage event showing or hiding weight columns.
+  $(window).bind('storage', $.proxy(function (e) {
```
Comment (indigoxela) [inline core/misc/tabledrag.js]: Function jQuery bind is also [deprecated](https://api.jquery.com/bind/).

File: core/misc/tabledrag.js (hunk 1 above)
Comment (klonos) [reply core/misc/tabledrag.js]: @indigoxela it seems that `.proxy()` was deprecated in jQuery v3.3, whereas we are on 1.x in Backdrop. Besides, it is used in many other .js files in core. Should we leave that to be part of https://github.com/backdrop/backdrop-issues/issues/3106 instead?

File: core/misc/tabledrag.js (hunk 2 above)
Comment (klonos) [reply core/misc/tabledrag.js]: Fixed with [`c5723f7` (#3870)](https://github.com/backdrop/backdrop/pull/3870/commits/c5723f7f78571004fac900e7c5eb4ed63ee5a248)

File: core/misc/tabledrag.js (hunk 1 above)
Comment (klonos) [reply core/misc/tabledrag.js]: Fixed with [`fdd72a4` (#3870)](https://github.com/backdrop/backdrop/pull/3870/commits/fdd72a40bc1c4c4bbe1c080b3c35a73d85b09c62)

File: core/misc/tabledrag.js [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
+/**
+ * Store the state of weight columns display for all tables.
+ *
+ * Default value is to hide weight columns.
+ *
+ * We are using JSON.parse here, because true/false get stored as strings.
+ * Otherwise showWeight would be either 'true' or 'false', and they would both
+ * be true in the toggle condition.
+ */
+var showWeight = JSON.parse(
```
Comment (indigoxela) [inline core/misc/tabledrag.js:14]: Any chance to avoid this global variable?

Comment (klonos) [reply core/misc/tabledrag.js:14]: Not sure @indigoxela 🤷🏼 ...my goal was to simply backport https://git.drupalcode.org/project/drupal/-/commit/148871a8c914c2db9c28ceac3ebf9763a0e83e7e

If there's a way to achieve that, then please feel free to suggest via PR against mine, or a completely new PR. Would be great to also repay our Drupal brethren by contributing the same back to Drupal.

**Comment by klonos** [comment]
Closing this in favor of https://github.com/backdrop/backdrop/pull/3869

**Pull request closed without merging**